src/lib.rs

```diff
@@ -376,12 +391,19 @@ impl StreamDeck {
             return Ok(());
         }
 
+        // MiraBoxM18 has 18 keys, but only the first 15 have screens
+        if self.kind == Kind::MiraBoxM18 && key >= 15 {
```

Comment:
Can you test with OpenDeck if pressing some of the "screenless" buttons turn display blank?

I've removed similar code from the AKP03* handling for this reason in #34

Reply:
I've digged more into this issue, and you also will need to move your check to `write_image` instead to avoid messing up the image cache